proposals/0000-provenance-qualified-imports.rst

```diff
@@ -102,6 +104,8 @@ Possible areas of disagreement/future work
 
 3) The potential of different choices for what is placed into the `source-origin` field and their meaning to tools like ``cabal-install`` is left open at this point. This will need to be worked out in the future to allow *fetching* from such specifications (i.e. when present in `cabal` files). A suggested heuristic would be as follows (in order of precedence): If the url is of a directory that contains a ``root.json`` it is assumed to be a package repository as defined in ``hackage-security``. If the url is of a ``.tar.gz`` file, it is a direct specification of a package tarball. If it terminates in ``.git``, it is assumed to be a git repo of a single package. (In this latter case, this leaves unresolved the question of a syntax for branches or tags, which would need to be determined). Because this proposal involves modifying the ``Cabal`` library, but not yet the ``cabal-install`` tool, we can afford to leave this not fully worked out for the time being.
 
+4) There is a question as to why ghc rather than e.g. cabal needs to know about source origin aliases. The answer is that the aliases are made reference to in haskell files. Meanwhile, the packagedb does not contain aliases, but contains the full, unabbreviated provenance url. Since it is ghc that needs to resolve the provenance qualified packages against the packagedb, then ghc itself needs to know the aliasing scheme.
```

Comment:
I must confess that I don't really understand the argument in this paragraph. The `source-origin` field is not meaningful, it is an identifier inserted by Cabal. What prevents Cabal from writing the alias in `source-origin`, rather than the full URL? My guess is that you have something in mind which would make it a bad choice. It is not yet, however, apparent in the proposal.